File: LCCI 08.06. Hanota/hanota.java
import java.util.*;
class hanota {
    public void solution(List<Integer> A, List<Integer> B, List<Integer> C) {
        int n = A.size();
        if (n == 0) {
        	return;
        }
        move(n, A, B, C);
    }

    public void move(int n, List<Integer> A, List<Integer> B, List<Integer> C) {
    	if (n == 1) {
    		C.add(A.remove(A.size() - 1));
    		return;
    	}
    	move(n - 1, A, C, B);
    	C.add(A.remove(A.size() - 1));
    	move(n - 1, B, A, C);
    }

    public static void main(String[] argv) {
    	List<Integer> A = new ArrayList<>();
    	A.add(2);
    	A.add(1);
    	A.add(0);
    	List<Integer> B = new ArrayList<>();
    	List<Integer> C = new ArrayList<>();
    	hanota obj = new hanota();
    	obj.solution(A, B, C);
    }
}

File: LCCI 08.06. Hanota/hanota.py
class hanota:
    def solution(self, A, B, C):
        n = len(A)
        if n == 0:
            return
        self.move(n, A, B, C)

    def move(self, n, A, B, C):
        if n == 1:
            C.append(A[-1])
            A.pop()
            return
        self.move(n - 1, A, C, B)
        C.append(A[-1])
        A.pop()
        self.move(n - 1, B, A, C)

obj = hanota()
A = [2, 1, 0]
B = []
C = []
obj.solution(A, B, C)